aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -71,12 +71,12 @@ void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
 
 void THTensor_(maskedFill)(THTensor *tensor, THByteTensor *mask, scalar_t value)
 {
-#ifdef _OPENMP
+#ifdef INTRA_OP_PARALLEL
   int64_t tensor_size = THTensor_(nElement)(tensor);
   int tensor_contig = THTensor_(isContiguous)(tensor);
   int mask_contig = THTensor_(isContiguous)(mask);
-  if (!omp_in_parallel() && tensor_contig && mask_contig) {
-    TH_TENSOR_APPLY2_OMP(tensor_size, tensor_contig, mask_contig,
+  if (!at::in_parallel_region() && tensor_contig && mask_contig) {
```

Comment:
Isn't this already handled by at::parallel_for itself? Or should we add this for future compatibility? I'd say it's up to the threadpool library whether or not it provides good support for nested parallelism.

Reply:
right, we seem to be doing this check twice, will update